pkg/agents/team6/treaties.go

```diff
@@ -0,0 +1,304 @@
+		default:
+			return consultUtility
+
+		}
+	}
+	return false
+}
+
+// food treaties
+func (a *CustomAgent6) considerFoodTreaty(conditionCheck bool, consultUtility bool, convertedFood food.FoodType) bool {
+	switch a.currBehaviour.string() {
+	case "Altruist":
+		return true
+	case "Collectivist":
+		if conditionCheck || convertedFood <= 2 {
+			return consultUtility
+		}
+		return true
+	case "Selfish":
+		if conditionCheck || convertedFood <= 60 {
+			return consultUtility
+		}
+		return true
+
+	default:
+		return consultUtility
+	}
+}
+
+// HP treaties
+func (a *CustomAgent6) considerHPTreaty(conditionCheck bool, consultUtility bool, levels levelsData, conditionValue int) bool {
+	switch a.currBehaviour.string() {
+	case "Altruist":
+		return true
+	case "Collectivist":
+		if conditionCheck || conditionValue < a.HealthInfo().WeakLevel {
+			return consultUtility
+		}
+		return true
+
+	case "Selfish":
+		if conditionCheck || conditionValue < levels.strongLevel {
+			return consultUtility
+		}
+		return true
+	default:
+		return consultUtility
+	}
+}
+
+// Decides if to accept or reject a treaty using utility. Used in "considerUtility" function above
+func (a *CustomAgent6) considerTreatyUsingUtility(t *messages.Treaty) bool {
+
+	// Log
+
+	// 1. Estimate the food intake of the proposed treaty
+
+	// Calculates how much food has been available on average on this floor
+	sum := food.FoodType(0)
+	for _, food := range a.shortTermMemory {
+		sum += food
+	}
+	averageFoodAvailable := float64(sum) / math.Max(float64(len(a.shortTermMemory)), 1.0)
+
+	// Converts different treaty "Request types" to a "food intake"
+	estimatedTakeFood := a.convertToTakeFoodAmount(averageFoodAvailable, t.Request(), t.RequestValue())
+	if estimatedTakeFood == -1 {
+		return false
+	}
+
+	// Checks the exact request condition. Only consider meaningful treaties (GE, GT)
+	if t.RequestOp() == messages.GE || t.RequestOp() == messages.GT /*t.ConditionOp() == messages.EQ || */ {
+		// The treaty is of the form "Take X (or less) food"
+
+		// 2. Calculate the agent's utility given different outcomes (accept or reject treaty)
+
+		// a. estimated expected utility when accepting the treaty
+		treatyTrustFactor := 1.0
+		treatyUtility := treatyTrustFactor * calculateUtility(float64(estimatedTakeFood), float64(a.desiredFoodIntake()), a.currBehaviour.string())
+
+		// b. estimated utility when rejecting the treaty
+		currentShortTermUtility := a.evaluateUtility(a.shortTermMemory) // estimated utility on the current floor
+		currentLongTermUtility := a.evaluateUtility(a.longTermMemory)   // estimated utility over the entire time in the tower
+
+		// benefit of signing the treaty (> 0 == beneficial)
+		shortTermBenefit := treatyUtility - currentShortTermUtility
+		longTermBenefit := treatyUtility - currentLongTermUtility
+
+		estimatedPeriod := int(a.reassignPeriodGuess)
+		estimatedTimeLeft := estimatedPeriod - len(a.shortTermMemory)
+		benefit := 0.0
```

Comment:
Would personally move the long term benefit stuff into the else statement since you don't need it unless you enter the else.

Reply:
nice spot - done :)

Reply:
good catch! we changed it